### ISSUE TITLE
test error of undefined symbol[Question]

### ISSUE THREAD

**Issue by Aliengmx** [MIC-DKFZ/nnDetection#20]
### :question: Question

python3.8
pytorch1.9.0 / 1.8.0
cuda11.1 /11.0
GCC 7.5.0
All install look successful, when I run python -c "import torch; import nndet._C; import nndet"
Here is the error,  
ImportError: /nnDetection/nndet/_C.cpython-38-x86_64-linux-gnu.so: undefined symbol: _Z8nms_cudaRKN2at6TensorES2_f

Is anybody has any idea about this error?

**Comment by blacksino** [comment]
I have met the same problem，how do u solve it

**Comment by mjq93** [comment]
I have met the same problem，how do u solve it

**Comment by czifan** [comment]
I have met the same problem, how do u solve it ...

**Comment by GabrielaProszczuk** [comment]
I have met the same problem, how do u solve it ...

**Comment by GabrielaProszczuk** [comment]
I have met the same problem, how do u solve it ...

**Comment by mibaumgartner** [comment]
This error is usually caused by wrong CUDA versions between PyTorch and the System. Here is a checklist:

1) Check that your GPU (NVIDIA) driver is installed: `nvidia-smi` (in the terminal) and torch has access to it `import torch; torch.cuda.is_available()` (inside python)
2) Check that CUDA is installed correctly: `nvcc --version`
3) Run `nndet_env` or `python utils.py` (inside the script directory of nnDetection) and make sure that `PyTorch CUDA` matches `Build cuda_` from the System Information

**Comment by GabrielaProszczuk** [comment]
Okay, I check all commands as follows and this are results. It seems like all versions are compatible?

1. `torch.cuda.is_available()` returns `True`
2.  CUDA version is `Cuda compilation tools, release 10.2, V10.2.89`
3. `nndet_env` returns PyTorch CUDA is also 10.2

This is the exact error showing:
`Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: /net/people/plgproszczuk/nnDetection/nndet/_C.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZN6caffe28TypeMeta21_typeMetaDataInstanceIdEEPKNS_6detail12TypeMetaDataEv
`

What else might be needed to be checked?

**Comment by mibaumgartner** [comment]
Newer GPUs (e.g. RTX 3000 series) are only supported from CUDA 11.1 +  which might be problem. Can you double check that the your GPU Architecture is supported by CUDA 10.2 (https://docs.nvidia.com/deeplearning/cudnn/support-matrix/index.html) ?

**Comment by GabrielaProszczuk** [comment]
Cluster I'm using is working on Nvidia Tesla K40 XL GPUs. I don't see them on this list, but by default I had 11.2 CUDA version installed and I changed to 10.2 to match nndet_env CUDA version. When it was 11.2 I had the same error. What can I do?

**Comment by mibaumgartner** [comment]
Did you make sure to delete the `build` folder in nnDetection after you switched the CUDA versions?
If not, please delete the `.egg` and the `build` folder and reinstall nnDetection (via `pip install -e . -v`) while having the correct CUDA version active. `build` is used as a cache and simply rerunning the pip command won't recompile the code.

**Comment by GabrielaProszczuk** [comment]
I did not do that, but now other errors show. It seems that while trying to reinstall wrong compiler is used.
I get an error:
`    Your compiler (icpc) is not compatible with the compiler Pytorch was
    built with for this platform, which is g++ on linux. Please
    use g++ to to compile your extension. Alternatively, you may
    compile PyTorch from source using icpc, and then you can also use
    icpc to compile your extension.`
In between both installations I didn't do anything with compilers. At least not directly? I loaded 10.2 CUDA module and installed some packages, i.a correct version of torch and torchversion, because didn't have 1.7+ version before.  I'm not sure how to compile from source or how to use different compiler.
